### DISCUSSION TITLE
“The Request signature we calculated does not match the signature you provided.” 403 when uploading to ente minio storage

### DISCUSSION THREAD

**Discussion by Sakooooo** [ente-io/ente#2032] [Q&A] [answered]
I am proxying the minio s3 storage container behind nginx and use it for museum to access it, uploading from my devices works and the s3 storage holds the photo but the museum container responds with 500 and says that minio has responded with a 403, running 
`mc admin trace -v myminio` outputs the error in the title when museum runs a request for the file size. 

I have searched for a fix for the error minio and have not found any.

**Comment by mnvr** [comment]
This sounds similar - https://github.com/minio/minio/discussions/15197#discussioncomment-3065840. The S3 bucket (minio) URL that is being used by museum to generate the presigned URL differs from what minio is expecting. You can try to compare the presigned URL and glance at it to make sure everything seems correct.

This is also similar to https://github.com/ente-io/ente/discussions/1847#discussioncomment-9552743, though in that case I don't know if OP ever managed to solve their issue.

**Reply by mnvr** [reply, marked as answer]
If you ask me, I think the issue is somewhere in this part - "minio s3 storage container behind nginx" - so my suggestion would be to first try and get it running without nginx or any other proxy in the way, and then move on to adding nginx in the mix.

**Reply by Sakooooo** [reply]
> If you ask me, I think the issue is somewhere in this part - "minio s3 storage container behind nginx" - so my suggestion would be to first try and get it running without nginx or any other proxy in the way, and then move on to adding nginx in the mix.

i've just tried to run this without the nginx proxy, the upload just worked on desktop and mobile! which is kind of weird?

ill go try again with the nginx proxy this time but for now this is probably the answer! thank you so much!

**Reply by Sakooooo** [reply]
have given up trying to put it behind a proxy, would always keep failing with 403

anyways the answer is to just not use a proxy and use your machine's ip! thanks for all your help though

**Reply by mnvr** [reply]
> the upload just worked on desktop and mobile! which is kind of weird?

That might mean it is an issue with CORS.

**Reply by Sakooooo** [reply]
> That might mean it is an issue with CORS.

tried to use CORS just now with minio proxied behind nginx, museum still gets a 403 clients upload the file successfully though

```
if ($request_method = 'OPTIONS') {
add_header 'Access-Control-Allow-Origin' '*';
add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS';
                        #
                        # Custom headers and headers various browsers *should* be OK with but aren't
                        #
add_header 'Access-Control-Allow-Headers' 'DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range';
                        #
                        # Tell client that this pre-flight info is valid for 20 days
                        #
add_header 'Access-Control-Max-Age' 1728000;
add_header 'Content-Type' 'text/plain; charset=utf-8';
add_header 'Content-Length' 0;
return 204;
}
if ($request_method = 'POST') {
add_header 'Access-Control-Allow-Origin' '*' always;
add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS' always;
add_header 'Access-Control-Allow-Headers' 'DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range' always;
add_header 'Access-Control-Expose-Headers' 'Content-Length,Content-Range' always;
}
if ($request_method = 'GET') {
add_header 'Access-Control-Allow-Origin' '*' always;
add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS' always;
add_header 'Access-Control-Allow-Headers' 'DNT,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range' always;
add_header 'Access-Control-Expose-Headers' 'Content-Length,Content-Range' always;
}

proxy_pass http://ente-minio:3200;
```
this is the part of my config in nginx for minio that enables CORS
